index.js

```diff
@@ -6,19 +6,55 @@ var AbstractIterator = require('abstract-leveldown').AbstractIterator
 var inherits = require('inherits')
 var Codec = require('level-codec')
 var EncodingError = require('level-errors').EncodingError
+var specialMethods = ['approximateSize', 'compactRange']
 
 module.exports = DB.default = DB
 
 function DB (db, opts) {
   if (!(this instanceof DB)) return new DB(db, opts)
-  AbstractLevelDOWN.call(this, '')
+  if (db.supports && db.supports.encodings) throw new Error('Double encoding')
+
+  AbstractLevelDOWN.call(this, db.supports)
+  this.supports.encodings = true
+
+  // TODO (future major): remove this fallback
+  specialMethods.forEach(function (m) {
+    if (typeof db[m] === 'function' && !this.supports.additionalMethods[m]) {
+      this.supports.additionalMethods[m] = {
+        args: [{ type: 'key' }, { type: 'key' }, { type: 'options' }]
```

Comment:
Note that if we commit to this format here, then `leveldown` and `rocksdb` must also use it (and going back on it will be hard).